src/syscall/syscall_libc.go

```diff
@@ -76,10 +95,28 @@ func splitSlice(p []byte) (buf *byte, len uintptr) {
 	return slice.buf, slice.len
 }
 
+// int open(const char *path, int oflag, int perm)
+//export open
+func libc_open(path *byte, oflag int, perm uint32) int
```

Comment:
`int` in C is practically always 32-bit (int32) on both 32-bit and 64-bit systems, while `int` in Go is 64-bit on 64-bit systems. Therefore, this signature is subtly wrong in an almost-compatible way. This is better:

```suggestion
func libc_open(path *byte, oflag int32, perm uint32) int32
```

Reply:
Where is a suitable place in the file tree for a test ? 
Can we use go test framework ?